lib/lte_link_control/lte_lc.c

```diff
@@ -346,6 +685,37 @@ int lte_lc_init_and_connect(void)
 	return err;
 }
 
+int lte_lc_connect_async(lte_lc_evt_handler_t handler)
+{
+	if (handler) {
+		evt_handler = handler;
+	} else if (evt_handler == NULL) {
+		LOG_ERR("No handler registered");
+		return -EINVAL;
+	}
+
+	return w_lte_lc_connect(false);
+}
+
+int lte_lc_init_and_connect_async(lte_lc_evt_handler_t handler)
+{
+	int err;
+
+	if (handler) {
+		evt_handler = handler;
+	} else if (evt_handler == NULL) {
+		LOG_ERR("No handler registered");
+		return -EINVAL;
+	}
+
+	err = w_lte_lc_init();
+	if (err) {
+		return err;
+	}
+
+	return w_lte_lc_connect(false);
```

Comment:
Why not just calling `lte_lc_connect_async` here after the init?
The `lte_lc_init_and_connect_async` function should call the `lte_lc_connect_async` to not duplicate code.
If the module is already initialized, it will not be initialized twice (see my other comment, a check is missing in the init function).